src/sentry/issues/priority.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+import logging
+from enum import Enum, IntEnum
+
+from sentry import features
+from sentry.models.activity import Activity
+from sentry.models.group import Group
+from sentry.models.grouphistory import GroupHistoryStatus, record_group_history
+from sentry.types.activity import ActivityType
+
+
+class PriorityLevel(IntEnum):
+    LOW = 25
+    MEDIUM = 50
+    HIGH = 75
+
+
+PRIORITY_LEVEL_TO_STR: dict[int, str] = {
+    PriorityLevel.LOW: "low",
+    PriorityLevel.MEDIUM: "medium",
+    PriorityLevel.HIGH: "high",
+}
+
+
+class PriorityChangeReason(Enum):
+    ESCALATING = "escalating"
+    ONGOING = "ongoing"
+
+
+PRIORITY_TO_GROUP_HISTORY_STATUS = {
+    PriorityLevel.HIGH: GroupHistoryStatus.PRIORITY_HIGH,
+    PriorityLevel.MEDIUM: GroupHistoryStatus.PRIORITY_MEDIUM,
+    PriorityLevel.LOW: GroupHistoryStatus.PRIORITY_LOW,
+}
+
+logger = logging.getLogger(__name__)
+
+
+def update_priority(
```

Comment:
cc @17hogeju, _hopefully_  we can reuse this for the PUT endpoint ticket

Reply:
Yeah this seems good to reuse, just need to remember include the `user` when the endpoints get used

Reply:
Will we add the logic to lock the priority etc here when we build the endpoint?